Neos.Neos/Classes/Controller/Service/NodesController.php

```diff
@@ -148,22 +158,28 @@ public function indexAction(
                 $entryNode = $subgraph->findNodeByAbsolutePath($nodePath);
             }
 
-            $nodes = !is_null($entryNode) ? $subgraph->findDescendantNodes(
-                $entryNode->aggregateId,
-                FindDescendantNodesFilter::create(
-                    nodeTypes: NodeTypeCriteria::create(
-                        NodeTypeNames::fromStringArray($nodeTypes),
-                        NodeTypeNames::createEmpty()
-                    ),
-                    searchTerm: $searchTerm,
-                )
-            ) : [];
+            if (!is_null($entryNode)) {
+                $nodes = $subgraph->findDescendantNodes(
+                    $entryNode->aggregateId,
+                    FindDescendantNodesFilter::create(
+                        nodeTypes: $nodeTypeCriteria,
+                        searchTerm: $searchTerm,
+                    )
+                );
+                if (
+                    SearchTermMatcher::matchesNode($entryNode, $searchTerm)
```

Comment:
If I read the code correctly, `$searchTerm` can indeed be `null` at this point leading to an exception..